src/client/voice/mod.rs

```diff
@@ -0,0 +1,338 @@
+/*
+ * voice.rs, LsangnaBoi 2022
+ * voice channel functionality
+ */
+use serenity::{
+    client::Context,
+    model::interactions::application_command::{ApplicationCommandInteraction, ApplicationCommandInteractionDataOptionValue},
+};
+use serenity::utils::Colour;
+use songbird::model::id::GuildId;
+use songbird::{input::ytdl_search, create_player};
+
+///join voice channel
+pub async fn join(ctx: &Context, command: &ApplicationCommandInteraction) -> String {
+    
+    //get cache and user ID
+    let cache = &ctx.cache;
+    let user_id = command.user.id;
+    let guild_id = command.guild_id;
+    
+    //create manager
+    let manager = songbird::get(ctx)
+        .await
+        .expect("Songbird voice client placed in at initialization");
+    
+    //retrieve the guild from the cache
+    if let Some(guild) = cache.guild(guild_id.unwrap()).await {
+
+        //get channel_id
+        let channel_id = guild
+            .voice_states
+            .get(&user_id)
+            .and_then(|voice_state| voice_state.channel_id);
+
+        let connect_to = match channel_id {
+            Some(channel) => channel,
+            None => {
+                return String::from("You must be in a voice channel to use this command!");
+            },
+        };
+
+        let name = channel_id.unwrap().name(cache).await.unwrap();
```

Comment:
would be great if expect was used instead of unwrap for better error messages, but that's just my personal preference